src/group_theory/group_action/support.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.set.pointwise
+
+/-!
+# Support of an element under an action action
+
+Given an action of a group `G` on a type `α`, we say that a set `s : set α ` supports an element
+`a : α` if, for all `g`, `g` fixes `a` if `g` fixes `s` pointwise.
+
+This is crucial in Fourier-Motzkin constructions.
+-/
+
+open_locale pointwise
+
+variables {G H α β : Type*}
+
+namespace mul_action
+section has_smul
+variables (G) [has_smul G α] [has_smul G β]
+
+/-- A set `s` supports `b` if `g • b = b` whenever `g • a = a` for all `a ∈ s`. -/
+@[to_additive "A set `s` supports `b` if `g +ᵥ b = b` whenever `g +ᵥ a = a` for all `a ∈ s`."]
+def supports (s : set α) (b : β) := ∀ g : G, (∀ ⦃a⦄, a ∈ s → g • a = a) → g • b = b
+
+variables {s t : set α} {a : α} {b : β}
+
+@[to_additive] lemma supports_of_mem (ha : a ∈ s) : supports G s a := λ g h, h ha
+
+variables {G}
+
+@[to_additive] lemma supports.mono (h : s ⊆ t) (hs : supports G s b) : supports G t b :=
+λ g hg, hs _ $ λ a ha, hg $ h ha
+
+end has_smul
+
+variables [group H] [has_smul G α] [has_smul G β] [mul_action H α] [has_smul H β]
+  [smul_comm_class G H β] [smul_comm_class G H α] {s t : set α} {b : β}
+
+-- TODO: This should work without `smul_comm_class`
+@[to_additive] lemma supports.smul (g : H) (h : supports G s b) : supports G (g • s) (g • b) :=
```

Comment:
This is the one lemma I care about and I can't afford `smul_comm_class` there. Thoughts? I can afford `is_scalar_tower` instead.

Reply:
Can you explain the context you want this to hold in?

Reply:
The context is complicated, but the gist of it is that we have a subgroup of `perm X` acting on a decorated version of `set X`, and we can get everything down to `perm X` acting on `set X` if needed. The full story is [here](https://github.com/leanprover-community/con-nf/blob/main/src/phase1/allowable.lean#L171-L173).

Reply:
I tried a few things and couldn't find an obvious set of hypotheses that would make this true. If you assume `is_scalar_tower`, would that involve `has_smul G H` or `has_smul H G` (or both?)

Reply:
Both IIRC

Reply:
I did a bit more experimenting and couldn't find any way to prove this without assuming `smul_comm`. Do you want to go ahead and PR this version, then later figure out a lemma that works for your usecase?

Reply:
Yeah, I see no point in waiting. The definition won't change.